src/www/ui_components/server_list.ts

```diff
@@ -0,0 +1,97 @@
+/*
+  Copyright 2021 The Outline Authors
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import {computed, customElement, property} from '@polymer/decorators';
+import {html, PolymerElement} from '@polymer/polymer';
+
+import {Server} from '../model/server';
+
+import {ServerCard} from './server_card';
+
+@customElement('server-list')
+export class ServerList extends PolymerElement {
```

Comment:
Let's convert only this element in this PR. The TS code can depend on the JS code (use the .js extension in the import), and even leverage the type guessing from TS. We can also add JSDoc type annotations like we do in the manager.

Example of TS depending on JS: https://github.com/Jigsaw-Code/outline-server/blob/f340c0d484514d306ce5f68a4bb5c7ec82d5a3b5/src/server_manager/web_app/ui_components/outline-gcp-oauth-step.ts

Example js with type annotations depending on both js and ts: https://github.com/Jigsaw-Code/outline-server/blob/f340c0d484514d306ce5f68a4bb5c7ec82d5a3b5/src/server_manager/web_app/ui_components/app-root.js

That was all before Ben converted the elements all to TS: https://github.com/Jigsaw-Code/outline-server/pull/997